## Carico librerie

In [20]:
import pandas as pd

import warnings
warnings.filterwarnings('ignore')

## Importo il dataset scaricato in formato csv

In [21]:

path_local_machine = 'C:/Users/User/Desktop/dati covid/'
file_name = 'owid-covid-data.csv'

df_covid = pd.read_csv(path_local_machine+file_name,sep = ',',header = 0)

## Visualizzo prime 10 righe del file

In [22]:
df_covid.sort_values('total_cases',ascending=False).head(10)

iso_code continent location        date  total_cases  new_cases  \
372490  OWID_WRL       NaN    World  2024-02-04  774493392.0    24217.0   
372487  OWID_WRL       NaN    World  2024-02-01  774469175.0        0.0   
372486  OWID_WRL       NaN    World  2024-01-31  774469175.0        0.0   
372485  OWID_WRL       NaN    World  2024-01-30  774469175.0        0.0   
372484  OWID_WRL       NaN    World  2024-01-29  774469175.0        0.0   
372483  OWID_WRL       NaN    World  2024-01-28  774469175.0    77810.0   
372488  OWID_WRL       NaN    World  2024-02-02  774469175.0        0.0   
372489  OWID_WRL       NaN    World  2024-02-03  774469175.0        0.0   
372478  OWID_WRL       NaN    World  2024-01-23  774391365.0        0.0   
372476  OWID_WRL       NaN    World  2024-01-21  774391365.0   101740.0   

        new_cases_smoothed  total_deaths  new_deaths  new_deaths_smoothed  \
372490            3459.571     7026534.0        82.0               11.714   
372487           11115.714     7026452.0         0.0               70.429   
372486           11115.714     7026452.0         0.0               70.429   
372485           11115.714     7026452.0         0.0               70.429   
372484           11115.714     7026452.0         0.0               70.429   
372483           11115.714     7026452.0       493.0               70.429   
372488           11115.714     7026452.0         0.0               70.429   
372489           11115.714     7026452.0         0.0               70.429   
372478           14534.286     7025025.0         0.0              127.143   
372476           14534.286     7025025.0       890.0              127.143   

        ...  male_smokers  handwashing_facilities  hospital_beds_per_thousand  \
372490  ...        34.635                   60.13                       2.705   
372487  ...        34.635                   60.13                       2.705   
372486  ...        34.635                   60.13                       2.705   
372485  ...        34.635                   60.13                       2.705   
372484  ...        34.635                   60.13                       2.705   
372483  ...        34.635                   60.13                       2.705   
372488  ...        34.635                   60.13                       2.705   
372489  ...        34.635                   60.13                       2.705   
372478  ...        34.635                   60.13                       2.705   
372476  ...        34.635                   60.13                       2.705   

        life_expectancy  human_development_index    population  \
372490            72.58                    0.737  7.975105e+09   
372487            72.58                    0.737  7.975105e+09   
372486            72.58                    0.737  7.975105e+09   
372485            72.58                    0.737  7.975105e+09   
372484            72.58                    0.737  7.975105e+09   
372483            72.58                    0.737  7.975105e+09   
372488            72.58                    0.737  7.975105e+09   
372489            72.58                    0.737  7.975105e+09   
372478            72.58                    0.737  7.975105e+09   
372476            72.58                    0.737  7.975105e+09   

        excess_mortality_cumulative_absolute  excess_mortality_cumulative  \
372490                                   NaN                          NaN   
372487                                   NaN                          NaN   
372486                                   NaN                          NaN   
372485                                   NaN                          NaN   
372484                                   NaN                          NaN   
372483                                   NaN                          NaN   
372488                                   NaN                          NaN   
372489                                   NaN                          NaN   
372478                

# QUESITO 1

### Dimensioni del dataset:

In [23]:
print('Numero di righe: %d'%(df_covid.shape[0]))
print('Numero di colonne: %d'%(df_covid.shape[1]))

Numero di righe: 376985
Numero di colonne: 67


### Nomi delle colonne:

In [24]:
list(df_covid.columns)

['iso_code',
 'continent',
 'location',
 'date',
 'total_cases',
 'new_cases',
 'new_cases_smoothed',
 'total_deaths',
 'new_deaths',
 'new_deaths_smoothed',
 'total_cases_per_million',
 'new_cases_per_million',
 'new_cases_smoothed_per_million',
 'total_deaths_per_million',
 'new_deaths_per_million',
 'new_deaths_smoothed_per_million',
 'reproduction_rate',
 'icu_patients',
 'icu_patients_per_million',
 'hosp_patients',
 'hosp_patients_per_million',
 'weekly_icu_admissions',
 'weekly_icu_admissions_per_million',
 'weekly_hosp_admissions',
 'weekly_hosp_admissions_per_million',
 'total_tests',
 'new_tests',
 'total_tests_per_thousand',
 'new_tests_per_thousand',
 'new_tests_smoothed',
 'new_tests_smoothed_per_thousand',
 'positive_rate',
 'tests_per_case',
 'tests_units',
 'total_vaccinations',
 'people_vaccinated',
 'people_fully_vaccinated',
 'total_boosters',
 'new_vaccinations',
 'new_vaccinations_smoothed',
 'total_vaccinations_per_hundred',
 'people_vaccinated_per_hundred',
 'peo

# QUESITO 2

In [25]:
#creo un df che contiene tutti i record che hanno il campo "continent" valorizzato
df_2 = df_covid[df_covid['continent'].notnull()]

#assunzione: considero i casi totali come somma giornaliera dei "new cases"

#creo un df che contiene tutti i record che hanno il campo "continent" valorizzato
df_2 = df_covid[df_covid['continent'].notnull()]
#sommo tutti i nuovi casi per ogni location all'interno del continente
df_2_grouped = df_2.groupby(['continent'])['new_cases'].sum().reset_index()
#rinomino colonna "new_cases" con "computed_total_cases"
df_2_grouped = df_2_grouped.rename(columns ={'new_cases':'computed_total_cases'})
df_2_grouped

continent  computed_total_cases
0         Africa            13136517.0
1           Asia           301391174.0
2         Europe           252013370.0
3  North America           124526003.0
4        Oceania            14741735.0
5  South America            68752780.0

# QUESITO 3

### Definisco la funzione del quesito 3:

In [41]:
def confronto_casi_totali_tra_continenti(df_covid, continente1, continente2):
    df_3_contienti = df_covid[df_covid['continent'].notnull()].groupby(['continent'])['new_cases'].sum().reset_index()
    df_3_contienti = df_3_contienti.rename(columns ={'new_cases':'computed_total_cases'})
    
    
    casi_totali_continente_1 = int(df_3_contienti[df_3_contienti['continent']==continente1]['computed_total_cases'])
    casi_totali_continente_2 = int(df_3_contienti[df_3_contienti['continent']==continente2]['computed_total_cases'])
    casi_totali_world = int(df_covid[df_covid['location']=='World']['new_cases'].sum())
    
    confronto_val_min = min(casi_totali_continente_1,casi_totali_continente_2)
    confronto_val_max = max(casi_totali_continente_1,casi_totali_continente_2)
    confronto_val_avg = (casi_totali_continente_1+casi_totali_continente_2)/2
    perc_1_vs_world = round(100*(casi_totali_continente_1/casi_totali_world),4)
    perc_2_vs_world = round(100*(casi_totali_continente_2/casi_totali_world),4)
    
    print('Numero casi totali %s: %d'%(continente1,casi_totali_continente_1))
    print('Numero casi totali %s: %d'%(continente2,casi_totali_continente_2))
    print('Numero casi totali mondo: %d'%(casi_totali_world))
    
    print('\n')
    print('Valore minimo: %d'%(confronto_val_min))
    print('Valore massimo: %d'%(confronto_val_max))
    print('Valore medio: %d'%(confronto_val_avg))    
    print('\n')
    print('percentuale casi %s vs World: %f'%(continente1,perc_1_vs_world)+' %')
    print('percentuale casi %s vs World: %f'%(continente2,perc_2_vs_world)+' %')
    

### Esempio:

In [42]:
#esempio: Asia vs Oceania
confronto_casi_totali_tra_continenti(df_covid,'Asia','Oceania')

Numero casi totali Asia: 301391174
Numero casi totali Oceania: 14741735
Numero casi totali mondo: 774561579


Valore minimo: 14741735
Valore massimo: 301391174
Valore medio: 158066454


percentuale casi Asia vs World: 38.911200 %
percentuale casi Oceania vs World: 1.903200 %


# QUESITO 4

In [43]:
def confronto_vaccinazioni_totali_tra_continenti(df_covid, continente1, continente2,continente3, continente4,continente5,continente6):
    df_3_contienti = df_covid[df_covid['continent'].notnull()].groupby(['continent'])['new_vaccinations'].sum().reset_index()
    df_3_contienti = df_3_contienti.rename(columns ={'new_vaccinations':'computed_total_vaccinations'})
    
    
    vaccinazioni_totali_continente_1 = int(df_3_contienti[df_3_contienti['continent']==continente1]['computed_total_vaccinations'])
    vaccinazioni_totali_continente_2 = int(df_3_contienti[df_3_contienti['continent']==continente2]['computed_total_vaccinations'])
    vaccinazioni_totali_continente_3 = int(df_3_contienti[df_3_contienti['continent']==continente3]['computed_total_vaccinations'])
    vaccinazioni_totali_continente_4 = int(df_3_contienti[df_3_contienti['continent']==continente4]['computed_total_vaccinations'])
    vaccinazioni_totali_continente_5 = int(df_3_contienti[df_3_contienti['continent']==continente5]['computed_total_vaccinations'])
    vaccinazioni_totali_continente_6 = int(df_3_contienti[df_3_contienti['continent']==continente6]['computed_total_vaccinations'])
    vaccinazioni_totali_world = int(df_covid[df_covid['location']=='World']['new_vaccinations'].sum())
    
    confronto_val_min = min(vaccinazioni_totali_continente_1,vaccinazioni_totali_continente_2,vaccinazioni_totali_continente_3, vaccinazioni_totali_continente_4,vaccinazioni_totali_continente_5,vaccinazioni_totali_continente_6)
    confronto_val_max = max(vaccinazioni_totali_continente_1,vaccinazioni_totali_continente_2,vaccinazioni_totali_continente_3, vaccinazioni_totali_continente_4,vaccinazioni_totali_continente_5,vaccinazioni_totali_continente_6)
    confronto_val_avg = (vaccinazioni_totali_continente_1+vaccinazioni_totali_continente_2+vaccinazioni_totali_continente_3+ vaccinazioni_totali_continente_4+vaccinazioni_totali_continente_5+vaccinazioni_totali_continente_6)/5
    perc_1_vs_world = round(100*(vaccinazioni_totali_continente_1/vaccinazioni_totali_world),4)
    perc_2_vs_world = round(100*(vaccinazioni_totali_continente_2/vaccinazioni_totali_world),4)
    perc_3_vs_world = round(100*(vaccinazioni_totali_continente_3/vaccinazioni_totali_world),4)
    perc_4_vs_world = round(100*(vaccinazioni_totali_continente_4/vaccinazioni_totali_world),4)
    perc_5_vs_world = round(100*(vaccinazioni_totali_continente_5/vaccinazioni_totali_world),4)
    perc_6_vs_world = round(100*(vaccinazioni_totali_continente_6/vaccinazioni_totali_world),4)
    
    print('Numero vaccinazioni totali %s: %d'%(continente1,vaccinazioni_totali_continente_1))
    print('Numero vaccinazioni totali %s: %d'%(continente2,vaccinazioni_totali_continente_2))
    print('Numero vaccinazioni totali %s: %d'%(continente3,vaccinazioni_totali_continente_3))
    print('Numero vaccinazioni totali %s: %d'%(continente4,vaccinazioni_totali_continente_4))
    print('Numero vaccinazioni totali %s: %d'%(continente5,vaccinazioni_totali_continente_5))
    print('Numero vaccinazioni totali %s: %d'%(continente6,vaccinazioni_totali_continente_6))
    
    print('Numero vaccinazioni totali mondo: %d'%(vaccinazioni_totali_world))
    
    print('\n')
    print('Valore minimo: %d'%(confronto_val_min))
    print('Valore massimo: %d'%(confronto_val_max))
    print('Valore medio: %d'%(confronto_val_avg))    
    print('\n')
    print('percentuale vaccinazioni %s vs World: %f'%(continente1,perc_1_vs_world)+' %')
    print('percentuale vaccinazioni %s vs World: %f'%(continente2,perc_2_vs_world)+' %')
    print('percentuale vaccinazioni %s vs World: %f'%(continente3,perc_3_vs_world)+' %')
    print('percentuale vaccinazioni %s vs World: %f'%(continente4,perc_4_vs_world)+' %')
    print('percentuale vaccinazioni %s vs World: %f'%(continente5,perc_5_vs_world)+' %')   
    print('percentuale vaccinazioni %s vs World: %f'%(continente6,perc_6_vs_world)+' %')   
    

### Esempio

In [44]:
#esempio: Asia vs Oceania
confronto_vaccinazioni_totali_tra_continenti(df_covid,'Asia','Oceania','Europe','Africa','North America', 'South America')

Numero vaccinazioni totali Asia: 7550421093
Numero vaccinazioni totali Oceania: 74815823
Numero vaccinazioni totali Europe: 1267550664
Numero vaccinazioni totali Africa: 117367074
Numero vaccinazioni totali North America: 1003616704
Numero vaccinazioni totali South America: 846374758
Numero vaccinazioni totali mondo: 13560669109


Valore minimo: 74815823
Valore massimo: 7550421093
Valore medio: 2172029223


percentuale vaccinazioni Asia vs World: 55.678800 %
percentuale vaccinazioni Oceania vs World: 0.551700 %
percentuale vaccinazioni Europe vs World: 9.347300 %
percentuale vaccinazioni Africa vs World: 0.865500 %
percentuale vaccinazioni North America vs World: 7.400900 %
percentuale vaccinazioni South America vs World: 6.241400 %


In [46]:
def confronto_casi_totali_tra_continenti_per_confronto(df_covid, continente1, continente2,continente3):
    df_3_contienti = df_covid[df_covid['continent'].notnull()].groupby(['continent'])['new_cases'].sum().reset_index()
    df_3_contienti = df_3_contienti.rename(columns ={'new_cases':'computed_total_cases'})
    
    
    casi_totali_continente_1 = int(df_3_contienti[df_3_contienti['continent']==continente1]['computed_total_cases'])
    casi_totali_continente_2 = int(df_3_contienti[df_3_contienti['continent']==continente2]['computed_total_cases'])
    casi_totali_continente_3 = int(df_3_contienti[df_3_contienti['continent']==continente3]['computed_total_cases'])
    casi_totali_world = int(df_covid[df_covid['location']=='World']['new_cases'].sum())
    
    confronto_val_min = min(casi_totali_continente_1,casi_totali_continente_2,casi_totali_continente_3)
    confronto_val_max = max(casi_totali_continente_1,casi_totali_continente_2,casi_totali_continente_3)
    confronto_val_avg = (casi_totali_continente_1+casi_totali_continente_2+casi_totali_continente_3)/3
    perc_1_vs_world = round(100*(casi_totali_continente_1/casi_totali_world),4)
    perc_2_vs_world = round(100*(casi_totali_continente_2/casi_totali_world),4)
    perc_3_vs_world = round(100*(casi_totali_continente_3/casi_totali_world),4)
    
    print('Numero casi totali %s: %d'%(continente1,casi_totali_continente_1))
    print('Numero casi totali %s: %d'%(continente2,casi_totali_continente_2))
    print('Numero casi totali %s: %d'%(continente3,casi_totali_continente_3))
    print('Numero casi totali mondo: %d'%(casi_totali_world))
    
    print('\n')
    print('Valore minimo: %d'%(confronto_val_min))
    print('Valore massimo: %d'%(confronto_val_max))
    print('Valore medio: %d'%(confronto_val_avg))    
    print('\n')
    print('percentuale casi %s vs World: %f'%(continente1,perc_1_vs_world)+' %')
    print('percentuale casi %s vs World: %f'%(continente2,perc_2_vs_world)+' %')
    print('percentuale casi %s vs World: %f'%(continente3,perc_3_vs_world)+' %')
    

In [48]:
confronto_casi_totali_tra_continenti_per_confronto(df_covid,'Europe','Oceania','South America')

Numero casi totali Europe: 252013370
Numero casi totali Oceania: 14741735
Numero casi totali South America: 68752780
Numero casi totali mondo: 774561579


Valore minimo: 14741735
Valore massimo: 252013370
Valore medio: 111835961


percentuale casi Europe vs World: 32.536300 %
percentuale casi Oceania vs World: 1.903200 %
percentuale casi South America vs World: 8.876300 %


# QUESITO 5

Andando a variare quello che è un Continente all'interno del confronto di cui sopra, possiamo ottenere i valori richiesti dei tre continenti: Sud America, Europa e Oceania.
Otteniamo quindi i seguenti risultati.

"Le cifre parlano chiaro: l'Europa, il Sud America e l'Oceania sono continenti con situazioni contrastanti nella battaglia contro il COVID-19. Mentre l'Europa registra il più alto numero di casi, seguito dal Sud America. Dall'altra parte, l'Oceania si distingue per la sua relativa bassa incidenza, tra i tre continenti analizzati risulta essere all'ultimo posto per numero di casi. Si rimanda alle statistiche sopra descritte un eventuale approfondimento quantitativo.

Ma c'è speranza nei dati delle vaccinazioni: l'Europa ha immunizzato il 9,34% del totale globale vaccinato, mentre il Sud America ha raggiunto il 6.24% del totale. Anche se l'Oceania ha una percentuale di vaccinati inferiore rappresenta comunque un passo avanti significativo nella lotta contro la pandemia, con lo 0.55% del totale globale vaccinato.